internal/coreinternal/processor/filterset/regexp/regexpfilterset.go

```diff
@@ -41,7 +42,19 @@ func NewFilterSet(filters []string, cfg *Config) (*FilterSet, error) {
 
 	if cfg != nil && cfg.CacheEnabled {
 		fs.cacheEnabled = true
-		fs.cache = lru.New(cfg.CacheMaxNumEntries)
+		var err error
+		if cfg.CacheMaxNumEntries == 0 {
+			// 0 means unlimited; since hashicorp/golang-lru does not allow for that,
+			// lets use the largest possible size
+			// TODO: Should we keep allowing such setting? (it does not seem to be well documented anyway)
+			fs.cache, err = lru.New(math.MaxInt)
```

Comment:
I'm afraid this can blow up memory utilization. I would rather disallow `0` value and introduce a reasonable default. 512 maybe?

Reply:
Yeah, I tried to reproduce the original behavior but was concerned with it as well. I think reasonable default makes a lot of sense

Reply:
I added a default, let me know what you think